### PR TITLE
AI - Add capital value for amphib attacks

### PR DESCRIPTION
**Pull request by ron-murhammer**
[Review on Reviewable](https://reviewable.io/reviews/triplea-game/triplea/316)

### CODE REVIEW

File: src/games/strategy/triplea/ai/proAI/ProCombatMoveAI.java [hunk 1]
```diff
@@ -660,11 +660,15 @@ private void removeTerritoriesWhereTransportsAreExposed() {
...
+        int isEnemyCapital = 0;
         final TerritoryAttachment ta = TerritoryAttachment.get(t);
         if (ta != null) {
           production = ta.getProduction();
+          if (ta.isCapital()) {
+            isEnemyCapital = 1;
+          }
         }
-        final double attackValue = result.getTUVSwing() + production;
+        final double attackValue = result.getTUVSwing() + production * (1 + 3 * isEnemyCapital);
```
Comment (DanVanAtta) [inline src/games/strategy/triplea/ai/proAI/ProCombatMoveAI.java]: - magic numbers code smell. The "1" and "3" are pretty meaningless to me, and likely will be to the author as well in a few months time

- lack of regression tests. It really does seem like you could mock out some of the key data structures and build tests around this logic. Then there would be confidence we are not hurting some other corner case.

File: src/games/strategy/triplea/ai/proAI/ProCombatMoveAI.java [hunk 2]
```diff
@@ -660,11 +660,15 @@ private void removeTerritoriesWhereTransportsAreExposed() {
             ProBattleUtils.calculateBattleResults(player, t, patd.getUnits(), patd.getMaxEnemyDefenders(player, data),
                 patd.getBombardTerritoryMap().keySet(), true);
         int production = 0;
+        int isEnemyCapital = 0;
         final TerritoryAttachment ta = TerritoryAttachment.get(t);
         if (ta != null) {
           production = ta.getProduction();
+          if (ta.isCapital()) {
+            isEnemyCapital = 1;
```
Comment (DanVanAtta) [inline src/games/strategy/triplea/ai/proAI/ProCombatMoveAI.java]: Would it not be simpler to just set this to 3, rather than using the 'magic number' scaling factor later? If the scaling factor were labelled as such, I would have less objection.

More importantly though, how do you know that this puts enough weight on launching amphib attacks on capitols? It does seem like we are missing the value of the IPC you would capture, and some more value for the strategic benefit of denying production for a turn as well.

Last, I think there should be yet even more weight added if the capitol is not recapturable. Given where things are, adding that consideration in the future I think is reasonable, but in the meantime I would tend to think some additional factors have not been added in to increase the weight even more.

Comment (ron-murhammer) [inline src/games/strategy/triplea/ai/proAI/ProCombatMoveAI.java]: Correct. I'm planning to refactor all territory/factory value code into a separate class as its somewhat scattered right now. And yes it should consider the PUs it would capture rather than just adding a multiple to the territory's value.

File: src/games/strategy/triplea/ai/proAI/ProCombatMoveAI.java (hunk 1 above)
Comment (ron-murhammer) [inline src/games/strategy/triplea/ai/proAI/ProCombatMoveAI.java]: I'll move it into the isEnemyCapital variable but to some extent these are arbitrary weights right now. Mostly just a stop gap so the AI doesn't bail on amphib attacking capitals.

**Comment by DanVanAtta** [comment]
Merging, updates and additions will be easier to review as their own PR/commits. For sure some more clarity could be put into the weights in "ProCombatMoveAI.java", but also I think ultimately spending some effort to test some of the key elements of logic is really the next thing that can be done. 

My concern is that without regression tests, we have so many cases where behavior should be one thing and not another, I am concerned that it will be very difficult to make changes without breaking something somewhere.
